README.md

```diff
@@ -66,18 +66,54 @@ To achieve this MLOS provides:
 
     Once hooks are created in the target system, iteration on the external agent can be more rapidly developed and deployed.
 
-## Build
+
+## Python only installation
+
+Some of the examples require only the installation of the mlos Python library. These examples do not use the shared memory infrastructure.
+To use this simplified installation, it's recommended to use the Anaconda python distribution and create a new conda environment:
+
+```
+$ conda create -n mlos_environment
+$ conda activate mlos_environment
+```
+
+You can download the mlos code using git:
+
+```
+$ git clone git@github.com:microsoft/MLOS.git
+```
+
+To install the Python library, change to the Python directory and install with ``pip``:
+
+```
+$ cd MLOS/source/Mlos.Python
+$ pip install -e .
+```
+
+You can also install the package directly without checking out the code:
+
+```
+$ pip install "git+https://github.com/microsoft/MLOS.git#egg=mlos&subdirectory=source/Mlos.Python"
+```
+
+However, this does not include the examples.
+
+After this installation, you can run any of the Python-only example notebooks.
+A good place to start is the [introduction to Bayesian Optimization](#).
+
+## Full Build (C# and C++ components)
 
 MLOS supports Windows and Linux build environments.
 
 For detailed instructions, please refer to:
 
-  1. [documentation/01-Prerequisites.md](./documentation/01-Prerequisites.md)
-  2. [documentation/02-Build.md](./documentation/02-Build.md)
+  1. [Prerequisites](./documentation/01-Prerequisites.md)
+  2. [Build](./documentation/02-Build.md)
 
 ## Examples
 
-Code and documentation for examples of using MLOS to optimize a system are described in [documentation/03-ExampleUsage.md](./documentation/04-ExampleUsage.md) and in the [source/Examples](./source/Examples/) tree.
+Code and documentation for examples of using MLOS to optimize a system are described in the [Notebooks](./notebooks/) section. Additional code is in the  [source/Examples](https://github.com/microsoft/MLOS/tree/main/source/Examples) source directory.
+You can find the source of the notebooks [on github as well](https://github.com/microsoft/MLOS/tree/main/source/Mlos.Notebooks).
```

Comment:
Do relative links not work here?

One issue with this is if they're browsing on one commit of the tree, then it could drop them to a different one.

Reply:
relative from where to where? this is rendered as a html on ``microsoft.github.io/mlos`` so I'm not sure where you'd want to link from there.

Reply:
This is the top level `README.md`, so if I for some reason end up at https://github.com/amueller/MLOS/tree/hugo_website, then if I click the "Prerequisites" link, I will end up on your version of that page still.  On the other hand the "source/Examples" link will now send me back to the original repo with a different revision.

Reply:
Yes. I'm just asking where else it should send you for the rendered page, because there is no relative paths to the source, right?

I can also remove this but then there is no link to github from the rendered pages.

Reply:
For the rendered page I think we can send them to the `main` branch version as you outlined here - that makes perfect sense to me since they're already navigating from an external source.

It was moreso can we do that (e.g. with more `sed` magic?) but still keep the relative links inside the git markdown?

Reply:
I don't see how we would do that. Maybe if the link ends with a ``/`` it should go to github? But that seems super brittle and we probably also need to start using a markdown parser if we want to do these kinds of rewrites.

Reply:
I think I see your point now.  A hackish way I could see to do it would be to append an anchor to the link and then regexp replace those strings specially.

For instance: 

```
[source/Examples/](./source/Examples/#mlos-github-tree-browse)
```

```sh
sed -r -e 's|\(\.(/[^#]+)#mlos-github-tree-browse\)|(https://github.com/microsoft/MLOS/tree/main/\1)/'
```

However, your point about it being brittle is well taken and this is perhaps not worth it, so I'm fine if you just want to leave these as is.

Reply:
the anchor is a neat idea, I was wondering how we could attach information to the link. But I think it's fine to leave as-is for now.